### PR TITLE
[ALLUXIO-2182] Display metrics for remote data acceleration

### PR DESCRIPTION
**Pull request by yupeng9**
https://alluxio.atlassian.net/browse/ALLUXIO-2182

This is a following up PR based on https://github.com/Alluxio/alluxio/pull/7251

image

It displays the metrics of short-circuit read, remote Alluxio remote as well as UFS read.

Note this PR introduced tagging concept, which is of pattern `metricName[.tagName:tagValue]*`. For example, the metric of `BytesReadUfs-UFS:${UFS}` becomes `BytesReadUfs.UFS:${UFS}`

### CODE REVIEW

**Comment by yupeng9** [comment]
Will recreate the PR targeting the `metrics-enhancement` branch.

**Comment by yupeng9** [comment]
Sweet, Github now can retarget the base branch.

**Comment by yupeng9** [comment]
Also add the cache hit percentage info

image

**Comment by yupeng9** [comment]
@aaudiber @calvinjia rebased on the merged `metrics-enhancement` branch.

File: core/common/src/main/java/alluxio/metrics/Metric.java [hunk 1]
```diff
@@ -13,21 +13,29 @@
...
   private static final long serialVersionUID = -2236393414222298333L;
 
   private static final String ID_SEPARATOR = "-id:";
+  public static final String TAG_SEPARATOR = ":";
   private final MetricsSystem.InstanceType mInstanceType;
   private final String mHostname;
   private final String mName;
   private final Long mValue;
   private String mInstanceId;
+  private final Map<String, String> mTags;
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/metrics/Metric.java]: Could we make a `MetricTag` object to make the associations between tag name and value more explicit?

File: core/common/src/main/java/alluxio/metrics/WorkerMetrics.java [hunk 2]
```diff
@@ -22,5 +22,10 @@
    */
   public static final String BYTES_READ_ALLUXIO = "BytesReadAlluxio";
 
+  /** Total number of bytes read from UFS through this worker. */
+  public static final String BYTES_READ_UFS = "BytesReadUfs";
+
+  public static final String UFS = "UFS";
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/metrics/WorkerMetrics.java]: Could we name this variable with some prefix to indicate it is a tag?

File: core/common/src/main/java/alluxio/metrics/Metric.java (hunk 1 above)
Comment (yupeng9) [reply core/common/src/main/java/alluxio/metrics/Metric.java]: I feel it's not necessary for now: tags are simply a map of string to string. I would consider a MetricTag object if it has other functionality added in the future.

**Comment by yupeng9** [comment]
@calvinjia PTAL

File: core/common/src/main/java/alluxio/metrics/Metric.java (hunk 1 above)
Comment (calvinjia) [reply core/common/src/main/java/alluxio/metrics/Metric.java]: It would make the code a lot easier to read, could you leave a TODO?

**Comment by calvinjia** [comment]
@yupeng9 Thanks for the update. I had one follow up request

File: core/common/src/main/java/alluxio/metrics/Metric.java (hunk 1 above)
Comment (yupeng9) [reply core/common/src/main/java/alluxio/metrics/Metric.java]: I can add a TODO, but I still think it adds unnecessary complexity. For example, the map provides checking existence and replace functionality. But an object of `tag` would require another index data structure for such functionality.

**Pull request merged**